src/components/plots/gridPlot.ts

```diff
@@ -48,23 +53,46 @@ export module Plot {
      */
     public project(attrToSet: string, accessor: any, scale?: Scale.AbstractScale<any, any>) {
       super.project(attrToSet, accessor, scale);
+
+      if (attrToSet === "x") {
+        super.project("x1", (d: any, i: number, u: any, m: Plot.PlotMetadata) => {
+          return scale.scale(this._projections["x"].accessor(d, i, u, m));
+        });
+      }
+
+      if (attrToSet === "y") {
+        super.project("y1", (d: any, i: number, u: any, m: Plot.PlotMetadata) => {
+          return scale.scale(this._projections["y"].accessor(d, i, u, m));
+        });
+      }
+
+      if (attrToSet === "x" && this._projections["x2"] === undefined) {
+        if (scale instanceof Scale.Category) {
+          super.project("x1", (d: any, i: number, u: any, m: Plot.PlotMetadata) => {
+            return scale.scale(this._projections["x"].accessor(d, i, u, m)) - scale.rangeBand() / 2;
+          });
+          super.project("x2", (d: any, i: number, u: any, m: Plot.PlotMetadata) => {
+            return scale.scale(this._projections["x"].accessor(d, i, u, m)) + scale.rangeBand() / 2;
+          });
+        }
+      }
+
+      if (attrToSet === "y" && this._projections["y2"] === undefined) {
+        if (scale instanceof Scale.Category) {
+          super.project("y1", (d: any, i: number, u: any, m: Plot.PlotMetadata) => {
+            return scale.scale(this._projections["y"].accessor(d, i, u, m)) - scale.rangeBand() / 2;
+          });
+          super.project("y2", (d: any, i: number, u: any, m: Plot.PlotMetadata) => {
+            return scale.scale(this._projections["y"].accessor(d, i, u, m)) + scale.rangeBand() / 2;
+          });
+        }
```

Comment:
While the above seems very sensible, I usually see this adjustments made at the generateAttrToProjector step.  That said, I'm not 100% certain where this adjustment should be made.